### PR TITLE
ADR for expose runner's machine info in log.

### PR DESCRIPTION
**Pull request by TingluoHuang**
💥  [Rendered](https://github.com/actions/runner/blob/users/tihuang/extralog/docs/adrs/0354-runner-machine-info.md) 💥 

https://github.com/actions/runner/issues/346

### CODE REVIEW

**Comment by hross** [comment]
Can detail be multiline? I'd prefer we just have one "group" that has all the info in it on multiple lines (but this way of doing it gives a lot of flexibility, which I like).

**Comment by TingluoHuang** [comment]
detail can be multiline, the runner handles post multiline string to log.

**Comment by TingluoHuang** [comment]
image

File: docs/adrs/0354-runner-machine-info.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# ADR 354: Expose runner machine info
+
+**Date**: 2020-03-02
+
+**Status**: Pending
+
+## Context
+
+- Provide an mechanism in the runner to include extra information in `Set up job` step's log.
```
Comment (hugovk) [inline docs/adrs/0354-runner-machine-info.md]:
```suggestion
- Provide a mechanism in the runner to include extra information in `Set up job` step's log.
```

File: docs/adrs/0354-runner-machine-info.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  Ex: Include OS/Software info from Hosted image.
+
+## Decision
+
+The runner will looking for a file `.extra_setup_info` under runner's root directory, the file could be a JSON with a simple schema.
```
Comment (hugovk) [inline docs/adrs/0354-runner-machine-info.md]:
```suggestion
The runner will look for a file `.extra_setup_info` under the runner's root directory, The file can be a JSON with a simple schema.
```

File: docs/adrs/0354-runner-machine-info.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  {
+    "group": "Software Detail",
+    "detail": "........"
+  }
+]
+```
+The runner will use `##[group]` and `##[endgroup]` to fold all detail info into an expandable group.
+
+Both [virtual-environments](https://github.com/actions/virtual-environments) and self-hosted runner can leverage this mechanism to add extra logging info to `Set up job` step's log.
```
Comment (hugovk) [inline docs/adrs/0354-runner-machine-info.md]:
```suggestion
Both [virtual-environments](https://github.com/actions/virtual-environments) and self-hosted runners can use this mechanism to add extra logging info to the `Set up job` step's log.
```

**Comment by TingluoHuang** [comment]
@chrispat @thejoebourneidentity 
Do you might check this ADR to make sure this change doesn't conflict with Product's future plan?

**Pull request merged**